code/game/objects/items/stacks/stack.dm

```diff
@@ -207,10 +207,33 @@
 			to_chat(usr, "<span class='warning'>There is another [R.title] here!</span>")
 			return FALSE
 
+		if(R.one_per_turf)
+			var/list/oneperturf = list(
+				/obj/structure/girder,
+				/obj/structure/falsewall,
+				/obj/structure/window,
+				/obj/structure/cult,
+				/obj/structure/table,
+				/obj/structure/chair,
+				/obj/structure/bed,
+				/obj/machinery/computer,
+				/obj/structure/computerframe,
+				/obj/machinery/door,
+				/obj/machinery/porta_turret
+			)
+			var/obj/S = locate() in get_turf(src)
+			if(is_type_in_list(S, oneperturf))
+				to_chat(usr, "<span class='warning'>There is already something here!</span>")
+				return FALSE
+
 		if(R.on_floor && !istype(get_turf(src), /turf/simulated))
 			to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")
 			return FALSE
 
+		if(R.on_floor && istype(get_turf(src), /turf/simulated/wall))
+			to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")
+			return FALSE
+
```

Comment:
This should be able to be combined with the above code block to form:
```suggestion
		if(R.on_floor && (!issimulatedturf(get_turf(src)) || iswallturf(get_turf(src)))
			to_chat(usr, "<span class='warning'>\The [R.title] must be constructed on the floor!</span>")
			return FALSE
```
Also switches over to use some `is_x` helpers. With this change, the block from 229-231 can be removed.